pkg/git/fetch.go

```diff
@@ -16,58 +30,87 @@ type Fetcher struct {
 }
 
 func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
-	f.Logger.Printf("Cloning %q @ %q...", gitURL, gitRevision)
+	f.Logger.Printf("Fetching %q @ %q...", gitURL, gitRevision)
+	auth, err := f.Keychain.Resolve(gitURL)
+	if err != nil {
+		return err
+	}
 
-	repository, err := git2go.InitRepository(dir, false)
+	tmpDir, err := os.MkdirTemp("", "git-clone-")
+	if err != nil {
+		return err
+	}
+
+	repository, err := gogit.PlainInit(tmpDir, false)
 	if err != nil {
 		return errors.Wrap(err, "initializing repo")
 	}
-	defer repository.Free()
 
-	remote, err := repository.Remotes.CreateWithOptions(parseURL(gitURL), &git2go.RemoteCreateOptions{
-		Name:  "origin",
-		Flags: git2go.RemoteCreateSkipInsteadof,
+	remote, err := repository.CreateRemote(&config.RemoteConfig{
+		Name: "origin",
+		URLs: []string{gitURL},
 	})
 	if err != nil {
 		return errors.Wrap(err, "creating remote")
 	}
-	defer remote.Free()
 
-	err = remote.Fetch([]string{"refs/*:refs/*"}, &git2go.FetchOptions{
-		DownloadTags: git2go.DownloadTagsAll,
-		RemoteCallbacks: git2go.RemoteCallbacks{
-			CredentialsCallback:      keychainAsCredentialsCallback(f.Keychain),
-			CertificateCheckCallback: certificateCheckCallback(),
-		},
-		ProxyOptions: git2go.ProxyOptions{
-			Type: git2go.ProxyTypeAuto,
-		},
-	}, "")
+	if httpsProxy, exists := os.LookupEnv("HTTPS_PROXY"); exists {
```

Comment:
Could this be extracted to a fun that returns a `transport.Transport` and sets it with `InstallProtocol`?